### PR TITLE
[API] Re-enable Doorkeeper::Grape::Helpers...

### PR DESCRIPTION
**Pull request by wildtangent**
...so authentication works correctly with the authorization code in the base class

I'm guessing you wanted to add some more work around the doorkeeper/OAuth functionality. For my purposes the ApiKey token is just fine...

This gets the API working again without too much hassle. You can pass in a param of `token=` or `X-Token` as a header

EDIT: 
* Added test coverage for Categories API controller
* Added test coverage for Docs API controller
* Changed `Simplecov.start` to `Simplecov.start 'rails'`

### CODE REVIEW

**Comment by scott** [comment]
I am super excited about these API contributions! I just invited you into the Slack room, hope to help answer any questions there...

**Comment by wildtangent** [comment]
hey, sorry! i messed up the commits the first time... forgot how to cherry pick ;)

Should be all atomic PR's now so won't overwrite each other...

File: app/controllers/api/v1/docs.rb [hunk 1]
```diff
@@ -68,10 +70,10 @@ class Docs < Grape::API
...
           requires :id, type: Integer, desc: "The ID of the Doc being updated"
-          requires :title, type: String, desc: "The name of the category of articles"
-          requires :category_id, type: Integer, desc: "The category the doc belongs to"
-          requires :body, type: String, desc: "The body/text of the article"
-          requires :user_id, type: Integer, desc: "The author of the article"
+          optional :title, type: String, desc: "The name of the category of articles"
+          optional :category_id, type: Integer, desc: "The category the doc belongs to"
+          optional :body, type: String, desc: "The body/text of the article"
+          optional :user_id, type: Integer, desc: "The author of the article"
           optional :keywords, type: String, desc: "Keywords that will be used for internal search and SEO"
```
Comment (scott) [inline app/controllers/api/v1/docs.rb]: I am curious why you switched this to optional when they are required by the model?  I will still merge it but just curious.

**Comment by wildtangent** [comment]
For an update though you the may not want to update all the properties.
Actually either way introduces a separate issue which is that should only
update values that were supplied not all the attributes, since this ends up
clearing the attributes not supplied which I didn't have time yet to
resolve.


On Friday, 1 July 2016, Scott Miller <notifications@github.com> wrote:

> Merged #208 <https://github.com/helpyio/helpy/pull/208>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/helpyio/helpy/pull/208#event-710517105>, or mute the
> thread
> <https://github.com/notifications/unsubscribe/AAAK6xYV1EdD-erQmlUZGctLbx0GD-RAks5qRRtEgaJpZM4JCOsK>
> .
>


-- 
*Joe Connor*
Freelance Developer

<http://factorymedia.com/>

*t: *0207 332 9700
*w: *factorymedia.com
*a: *primeandfire.com

*Factory Media UK*
Bike Magic <http://bikemagic.com/> | BMX Talk <http://bmxtalk.com/> | Cooler
<http://cooler.mpora.com/> | Dirt <http://dirt.mpora.com/> | Kingpin
<http://kingpin.mpora.com/> | MPORA <http://mpora.com/> | Onboard
<http://onboard.mpora.com/> | Outside Times <http://outsidetimes.com/> |
RCUK <http://roadcyclinguk.com/> | Ride UK BMX <http://rideukbmx.mpora.com/>
 | Sidewalk <http://sidewalk.mpora.com/> | Ski Union <http://skiunion.com/>
 | Surf Europe <http://surfeurope.mpora.com/> | Total Women's Cycling
<http://totalwomenscycling.com/>

*Factory Media GmbH*
Dirt <http://dirt.mpora.de/> | freedombmx <http://freedombmx.mpora.de/> |
MPORA <http://mpora.de/> | Monster Skateboard Magazine
<http://skateboardmsm.mpora.de/> | Skiing <http://skiingmag.mpora.de/>
| Snowboarder
MBM <http://snowboardermbm.mpora.de/> | Surfers <http://surfers.de> | *Total
Women's Cycling*

**Pull request merged**